### DISCUSSION TITLE
Why does fresh-start require sudo?

### DISCUSSION THREAD

**Discussion by fracz** [evertramos/nginx-proxy-automation#315] [Q&A] [answered]
Fresh clone of this project, trying to run `fresh-start.sh` and it immediately asks me for sudo password. Why? My user is in `docker` group so I'm able to manage docker containers. I do not have root access.

```
bin/fresh-start.sh 
⚡ Start execution '/home/xxx/proxy/bin/fresh-start.sh [sudo] password for xxx: 
Sorry, try again.
[sudo] password for xxx: 
sudo: 1 incorrect password attempt
[sudo] password for xxx: 
[sudo] password for xxx: 
[sudo] password for xxx: 
 ✔ success Function 'check_local_env_file'
[sudo] password for xxx: 
[sudo] password for xxx: 
[sudo] password for xxx: 
 ✔ success Function 'check_docker'
[sudo] password for xxx: 
[sudo] password for xxx:
```

It is even impossible to quit the script execution because after every Ctrl+C it asks me for sudo password again.

**Comment by evertramos** [comment, marked as answer]
Hello @fracz!

Thank you very much for your question.

This was in fact something to look at. We have a sudoless options coming up for [basescript](https://github.com/evertramos/basescript) and I was missing one function which I fixed today ([issue 11](https://github.com/evertramos/basescript/issues/11)).

If you please try to run again and it will only ask for 'sudo' if you don't have permission to write in the folders you are putting the nginx-proxy 'data' folder.

Also make sure to have 'false' in [*ALLOW_RUN_WITH_SUDO*](https://github.com/evertramos/nginx-proxy-automation/blob/d48b2477ed28cbda37738046079e35df219ba3e9/.env.sample#L20) variable, which is de default mode for nginx-proxy-automation.

Here is the update in basescript:

https://github.com/evertramos/basescript/commit/5c7de6e76e07e52908dde415e1ab0622011a48b2

Please let me know if you need any further help with that.

Best regards.
